epoch.go

```diff
@@ -1691,15 +1777,31 @@ func (e *Epoch) processNotarizedBlock(block Block, notarization *Notarization) e
 			return nil
 		}
 
+		// if we haven't timed out on the round, send a finalized vote message
```

Comment:
why do we need to do this? I don't think this is mandatory. We re-broadcast dropped finalized votes if we have a tail of notarized blocks that's not finalized. We will eventually re-broadcast this after we finish replication, if it's needed.